src/merkle/error.rs

```diff
@@ -2,7 +2,7 @@ use thiserror::Error;
 
 use super::{NodeIndex, RpoDigest};
 
-#[derive(Debug, Error)]
+#[derive(Clone, Debug, Error, Eq, PartialEq)]
 pub enum MerkleError {
```

Comment:
Why do we need to add these? I believe these were removed as a part of https://github.com/0xPolygonMiden/crypto/pull/344 to standardize how we handle errors.

Reply:
It seems that it was a solution some errors happening while changing the dependencies.
I have reverted these changes now.

Reply:
Should we revert these in a separate PR?

Reply:
I think you've already reverted these. If there are errors happening because of the missing derives in `miden-vm`, we should fix them there.